hollow-jar/full/src/main/java/org/wildfly/swarm/ts/hollow/jar/full/ProcessResult.java

```diff
@@ -0,0 +1,26 @@
+package org.wildfly.swarm.ts.hollow.jar.full;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Named;
+
+@Named
+@ApplicationScoped
+public class ProcessResult {
+
+    private volatile List<Object> writtenItems = new ArrayList<Object>();
+
+    public synchronized List<Object> getWrittenItems() {
```

Comment:
The way you're doing synchronization here is useless. Either all methods accessing `writtenItems` need to be `synchronized`, or none of them. Also, you're iterating on the content of the collection in other classes, and that would have to be synchronized as well. Do we really have multiple threads accessing the `ProcessResult`? If no, let's avoid synchronization altogether. If yes, maybe we could use something like `CopyOnWriteArrayList`.

The `writtenItems` field doesn't really have to be `volatile`.

Reply:
Yes I forgot remove this part before commit sorry :-(

Reply:
This is fixed when using await() so we avoid concurrent access.